Library/Homebrew/cmd/config.rb

```diff
@@ -119,9 +119,15 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
-    java =~ /java version "(.+?)"/ ? $1 : java
+  def describe_java
+    if which("java").nil?
+      "N/A"
+    elsif quiet_system("/usr/libexec/java_home --failfasti") && !$?.success?
```

Comment:
> failfasti oops!

Sorry, vim user here.

> Still want to keep the2>&1

Doesn't the `quiet_system` redirect both stdout and stderr?